app/services/project_life_cycle_steps/update_service.rb

```diff
@@ -48,7 +48,7 @@ def reschedule_following_phases
     end
 
     def initial_reschedule_date
-      model.active? ? model.finish_date + 1 : model.start_date
+      model.active? ? Day.next_working(from: model.finish_date).date : model.start_date
```

Comment:
I think it is better to handle finding next working day in `RescheduleService`

Also looking at `Day.next_working` I have a feeling that it is easy for it to return `nil` - it is not rare to have even few weeks of non working days on organisation level (like [CERN](https://home.cern/official-holidays#:~:text=New%20Year%E2%80%99s%20Eve-,Annual%20closure,-of%20the%20site))

Reply:
Unfortunately it's not trivial to unify all the `initial_reschedule_date` methods in the `RescheduleService`, because the [apply_working_days_change_job.rb#L37-L40](https://github.com/opf/openproject/blob/d4e94ae8d00fbd79788fe74d3e3da1429ed94271/app/workers/projects/phases/apply_working_days_change_job.rb#L37-L40) uses a different logic which does not require finding the next working day, so moving the initial schedule date is not really a feasible option.

Your concern about non-working days longer than a week is a valid point. However it is not directly related to this PR and  I would rather suggest logging a bug ticket for it to be addressed separately.